addons/project/controllers/portal.py

```diff
@@ -373,8 +373,13 @@ def portal_my_tasks(self, page=1, date_begin=None, date_end=None, sortby=None, f
         if search and search_in:
             domain += self._task_get_search_domain(search_in, search)
 
+        Task = request.env['project.task']
+        if not request.env.su:
+            domain = AND([domain, request.env['ir.rule']._compute_domain(Task._name, 'read')])
+        Task = Task.sudo()
```

Comment:
I think you can directly do this:
```suggestion
        Task = request.env['project.task'].sudo()
        domain = AND([domain, request.env['ir.rule']._compute_domain(Task._name, 'read')])
```
Because in this route we could assume the current user is not a public user since the route needs an authenticated user. 🙂 

However, in `/my/projects/<project_id>`, we display a list of tasks related to the project, so we also need this fix [there](https://github.com/odoo/odoo/blob/591d24dfdedc2358f615f7a35a4b14a5a5c1c30f/addons/project/controllers/portal.py#L59)

Also, can you rename `Task` into `TaskSudo` to indicate in the variable name, we are in sudo when we will call a method with this variable. 🙂